src/Neo/SmartContract/ContractParameter.cs

```diff
@@ -162,28 +162,22 @@ private static JObject ToJson(ContractParameter parameter, HashSet<ContractParam
                         json["value"] = parameter.Value.ToString();
                         break;
                     case ContractParameterType.Array:
-                        if (context is null)
-                            context = new HashSet<ContractParameter>();
-                        else if (context.Contains(parameter))
-                            throw new InvalidOperationException();
-                        context.Add(parameter);
+                        context ??= [];
+                        if (!context.Add(parameter)) throw new InvalidOperationException("Circular reference.");
                         json["value"] = new JArray(((IList<ContractParameter>)parameter.Value).Select(p => ToJson(p, context)));
-                        context.Remove(parameter);
+                        if (!context.Remove(parameter)) throw new InvalidOperationException("Circular reference.");
                         break;
                     case ContractParameterType.Map:
-                        if (context is null)
-                            context = new HashSet<ContractParameter>();
-                        else if (context.Contains(parameter))
-                            throw new InvalidOperationException();
-                        context.Add(parameter);
+                        context ??= [];
+                        if (!context.Add(parameter)) throw new InvalidOperationException("Circular reference.");
                         json["value"] = new JArray(((IList<KeyValuePair<ContractParameter, ContractParameter>>)parameter.Value).Select(p =>
                         {
                             JObject item = new();
                             item["key"] = ToJson(p.Key, context);
                             item["value"] = ToJson(p.Value, context);
                             return item;
                         }));
-                        context.Remove(parameter);
+                        if (!context.Remove(parameter)) throw new InvalidOperationException("Circular reference.");
```

Comment:
Why is this `if` statement needed?

Reply:
It should not happen, but always is a good practice checks the methods returns